remove run_events.trip_service_id from spec and trim examples

skyqrose opened the pull request:
- Removed the `trip_service_id` field.
- Rewrote most of the prose.
- Removed 3 examples.

Here are the 5 examples that were there before:
- Extra staffing for a special event
  - Kept, minor edits.
- Variations of runs by day of the week
  - Removed.
  - It doesn't show much more about the feature than the next example, and was way more complicated. I left a hint to the idea in the re-written spec text (about different runs on Monday vs Tuesday).
- Different runs mapping to the same set of trips
  - Kept, minor edits.
  - Added diagram.
- Jobs of entirely nonrevenue operations
  - Removed.
  - Not related to the `trip_service_id` idea.
  - Is related to TODS calendars, so could be useful to keep something like this outside the crew scheduling header.
  - But I couldn't find a way to reframe it that felt like scheduled data rather than something that would be done in realtime.
  - And it's not too complicated people should be able to figure it out from deadheads and `calendar_dates_supplement.txt` if they're motivated to represent this in their schedules, so I didn't think it was important to keep.
- Activating a storm schedule.
  - Removed.
  - This is a useful idea, but I think it's more of a GTFS concept, I don't think there's TODS-specific stuff here.
  
It may be easier to review this by looking at the diff to upstream/main, instead of the diff into your branch: [diff](https://github.com/cal-itp/operational-data-standard/compare/main...skyqrose:tods:80)

When this is merged, it should automatically show up in https://github.com/cal-itp/operational-data-standard/pull/80

skyqrose committed:
remove run_events.trip_service_id from spec and trim examples

jeffkessler-keolis commented on docs/spec/examples.md:
```diff
@@ -201,134 +203,41 @@ gameday,20240827,1
+service_id,run_id,event_sequence,block_id,job_type,event_type,trip_id,start_location,start_time,end_location,end_time
+weekday,1,1,       ,collector,sign-in        ,   ,main_terminal,14:00:00,main_terminal,14:15:00
+weekday,1,2,BLOCK-A,collector,collector      ,101,main_terminal,14:45:00,ballpark     ,15:30:00
+gameday,2,1,       ,collector,sign-in        ,   ,main_terminal,14:00:00,main_terminal,14:15:00
```
This is valid, but I'd prefer to have these _also_ use `run_id` `1` to reiterate that `run_id` is non-unique individually, but rather must be unique on `(service_id, run_id)`.

jeffkessler-keolis commented on docs/spec/examples.md:
```diff
@@ -171,28 +171,30 @@ weekday,10000,30,BLOCK-A,deadhead       ,,stop-1,12:00:00,garage,12:10:00
-#### `run_events.txt`
+#### `calendar.txt`
 
-_Note that the `run_id` of 1 maps to the combination of `(service_id, trip_service_id)`, meaning other runs with the same `run_id` of `1` can be in effect at the same time provided they existin other combinations of `(service_id, trip_service_id)`._
```
See comment below, but I think this is an important note to reiterate; perhaps this should be moved below, though?

The key thing I want to avoid is a consumer thinking the feed is broken because there are two versions of `run_id` = `1`, each of which have different `service_id` values, though.

jeffkessler-keolis commented on docs/spec/examples.md:
````diff
@@ -337,100 +246,27 @@ To detail the presence of the new `service_id`s and assign them to their applica
-
-A special track inspection train is being operated on a particular day, supported by an extra crew position.
-
-_Note: The deadhead trips themselves could be defined in their own new `service_id` even without definining corresponding run data in `run_events.txt` using the `calendar_supplement.txt` or `calendar_dates_supplement.txt` files._
-
-#### `trips_supplement.txt`
-
-```csv
-route_id,service_id,trip_id,TODS_trip_type,direction_id
-line1,inspection_train,inspection_line1_ob,deadhead,0
-line1,inspection_train,inspection_line1_ib,deadhead,1
-line1,inspection_train,inspection_line2_ob,deadhead,0
-line1,inspection_train,inspection_line2_ib,deadhead,1
-line1,inspection_train,inspection_line3_ob,deadhead,0
-line1,inspection_train,inspection_line3_ib,deadhead,1
-```
-
-#### `stop_times_supplement.txt`
-
-```csv
-trip_id,arrival_time,stop_id
-inspection_line1_ob,downtown,08:00:00
-inspection_line1_ob,anytown,09:00:00
-inspection_line1_ib,anytown,09:15:00
-inspection_line1_ib,downtown,10:15:00
-inspection_line2_ob,downtown,11:00:00
-inspection_line2_ob,busyville,12:30:00
-inspection_line2_ib,busyville,13:00:00
-inspection_line2_ib,downtown,14:30:00
-inspection_line3_ob,downtown,15:00:00
-inspection_line3_ob,centerton,15:45:00
-inspection_line3_ib,centerton,16:00:00
-inspection_line3_ib,downtown,16:45:00
-```
-
-#### `calendar_dates_supplement.txt`
-
-```csv
-service_id,date,exception_type
-inspection_train,20240901
````
I do like the idea of including this as an example, because I think this is a big use case for the distinct TODS-specific modeling of non-public trips. That said, we could certainly cut this down to two trips of two stop_times each to simplify things. But open to cutting if the rest of the working group believes it to not be worthwhile.

jeffkessler-keolis commented on docs/spec/examples/different-runs-same-trips.png:
Just have to say, this is ***amazingly*** useful. Thank you!

skyqrose committed:
add back nonrevenue job example

skyqrose committed:
add more prominent note about non-unique run ids

skyqrose commented:
As discussed in another channel:

I added another note about the run id not being unique. It was already mentioned but not at all prominently. I don't want to add it to the example because it would distract from the complex situation that the example is trying to show.

I've added back the nonrevenue job, simplified and moved outside of the mismatched crew/vehicle schedule section.
